metrics-core/src/main/java/software/amazon/swage/metrics/record/file/FileRecorder.java

```diff
@@ -141,6 +141,10 @@ protected void record(
             return;
         }
 
+        if (!isValid(label)) {
+            throw new IllegalArgumentException("Invalid metric name: '" + label + "'");
```

Comment:
Where is this exception caught? 

I don't think that an error in recording a metric should cause the application to fail. It is one thing to throw an exception when configuring your application, such as not providing the a cloudwatch implementation to the cloudwatch recorder. However, failing to record a metric should not result in the application code failing. 
In this example recorder,  I think it is better to set the precedence of logging a warning and letting the application continue.